app.py

```diff
@@ -11,700 +11,306 @@
-            new_card_body,
-            "sched",
-        )
-
-    else:
-        return no_update, no_update, no_update
+def disp_initial_sched(
+    num_employees: int, rand_seed: int
+) -> tuple[DataFrame, DataFrame, bool, str]:
+    """Display initial availability schedule.
+
+    Display initial schedule in, and switch to, the availability
+    tab if number of employees or seed is changed.
+    """
+    # one less to account for trainee
+    num_employees -= 1
+
+    shifts = [str(i + 1) for i in range(NUM_DAYS_IN_MONTH)]
+    df = utils.build_random_sched(num_employees, shifts, rand_seed)
 
+    init_availability_table = utils.display_availability(df, NOW.month, NOW.year)
+    return (
+        init_availability_table,
+        init_availability_table,
+        True,  # disable the shedule tab when changing parameters
+        "availability-tab",  # jump back to the availability tab
+    )
+
+
+@app.long_callback(
+    Output("schedule-content", "children", allow_duplicate=True),
+    Output("schedule-tab", "disabled", allow_duplicate=True),
+    inputs=[
+        Input("run-button", "n_clicks"),
+        State("shifts-per-employee-select", "value"),
+        State("employees-per-shift-select", "value"),
+        State("checklist-input", "value"),
+        State("consecutive-shifts-select", "value"),
+        State("availability-content", "children"),
+    ],
+    running=[
+        # show cancel button and hide run button, and disable and animate results tab
+        (Output("cancel-button", "style"), {"display": "inline-block"}, {"display": "none"}),
+        (Output("run-button", "style"), {"display": "none"}, {"display": "inline-block"}),
+        # switch to schedule tab while running
+        (Output("schedule-tab", "disabled"), False, False),
+        (Output("tabs", "value"), "schedule-tab", "schedule-tab"),
+    ],
+    cancel=[Input("cancel-button", "n_clicks")],
+    prevent_initial_call=True,
+)
+def run_optimization(
+    run_click: int,
+    shifts_per_employee: list[int],
+    employees_per_shift: list[int],
+    checklist: list[int],
+    consecutive_shifts: int,
+    sched_df: DataFrame,
+) -> tuple[DataFrame, bool]:
+    """Run a job on the hybrid solver when the run button is clicked."""
+    if run_click == 0 or ctx.triggered_id != "run-button":
+        raise PreventUpdate
+
+    shifts = list(sched_df["props"]["data"][0].keys())
+    shifts.remove("Employee")
+    availability = utils.availability_to_dict(sched_df["props"]["data"], shifts)
+    employees = list(availability.keys())
+
+    isolated_days_allowed = True if 0 in checklist else False
+    manager_required = True if 1 in checklist else False
+
+    cqm = employee_scheduling.build_cqm(
+        availability,
+        shifts,
+        *shifts_per_employee,
+        *employees_per_shift,
+        manager_required,
+        isolated_days_allowed,
+        consecutive_shifts + 1,
+    )
+
+    feasible_sampleset, errors = employee_scheduling.run_cqm(cqm)
```

Comment:
Might be for a different PR but should we do something with `errors` here?

Reply:
Yeah, they were shown before, and it might make sense to show them somewhere still (perhaps in another hidden div). I'll leave that for the next PR though.